finality-aleph/src/justification/requester.rs

```diff
@@ -78,73 +75,66 @@ where
         } = notification;
 
         if number <= last_finalized || number > stop_h {
-            debug!(target: "afa", "Not finalizing block {:?}. Last finalized {:?}, stop_h {:?}", number, last_finalized, stop_h);
+            debug!(target: "aleph-justification", "Not finalizing block {:?}. Last finalized {:?}, stop_h {:?}", number, last_finalized, stop_h);
             return;
         };
 
         if !(verifier.verify(&justification, hash)) {
-            warn!(target: "afa", "Error when verifying justification for block {:?} {:?}", number, hash);
+            warn!(target: "aleph-justification", "Error when verifying justification for block {:?} {:?}", number, hash);
             return;
         };
 
-        debug!(target: "afa", "Finalizing block {:?} {:?}", number, hash);
+        debug!(target: "aleph-justification", "Finalizing block {:?} {:?}", number, hash);
         let finalization_res = self.finalizer.finalize_block(
             hash,
             number,
             Some((ALEPH_ENGINE_ID, justification.encode())),
         );
         match finalization_res {
             Ok(()) => {
-                self.justification_request_delay.on_block_finalized();
-                debug!(target: "afa", "Successfully finalized {:?}", number);
+                self.justification_request_scheduler.on_block_finalized();
+                debug!(target: "aleph-justification", "Successfully finalized {:?}", number);
                 if let Some(metrics) = &self.metrics {
                     metrics.report_block(hash, Instant::now(), Checkpoint::Finalized);
                 }
             }
             Err(e) => {
-                error!(target: "afa", "Fail in finalization of {:?} {:?} -- {:?}", number, hash, e);
+                error!(target: "aleph-justification", "Fail in finalization of {:?} {:?} -- {:?}", number, hash, e);
             }
         }
     }
 
-    fn maintain_requests(&mut self) {
-        if self.client.info().finalized_number == self.last_seen_finalized {
-            self.attempt += 1;
-        } else {
-            self.attempt = 0;
-            self.last_seen_finalized = self.client.info().finalized_number;
-        }
-
-        if self.attempt == MAX_ATTEMPS {
-            self.block_requester.clear_justification_requests();
-            self.attempt = 0;
-        }
-    }
+    pub fn request_justification(&mut self, num: NumberFor<B>) {
+        match self.justification_request_scheduler.schedule_action() {
+            SchedulerActions::Request => {
+                let num = if num > self.client.info().best_number
+                    && self.client.info().best_number > self.min_allowed_delay.into()
+                {
+                    self.client.info().best_number - self.min_allowed_delay.into()
```

Comment:
The only uses of `min_allowed_delay` call `into` on it – you should probably define it as being of the appropriate type in the first place?

Reply:
Actually it is appropriate type, I just forgot to delete `into` s

Reply:
Anyway - Done.